### PR TITLE
[TS] fix: theme types of Button

### PR DESCRIPTION
**Pull request by ShimiSun**
#### What does this PR do?
It fixes the theme type of button.

#### Where should the reviewer start?
base.d.ts
#### What testing has been done on this PR?
local environment, tests 
#### How should this be manually tested?
review the code
#### Do Jest tests follow these best practices?
N/A

#### Any background context you want to provide?
N/A

#### What are the relevant issues?
Follow up to a PR #5651 (I think) 
#### Do the grommet docs need to be updated?
Nope, it's all bugs.
#### Should this PR be mentioned in the release notes?
Yes, as a TS fix
#### Is this change backward compatible or is it a breaking change?
backward compatible

### CODE REVIEW

File: src/js/themes/base.d.ts [hunk 1]
```diff
@@ -195,7 +196,13 @@ interface ButtonType {
...
       })
     | { [key: string]: ButtonKindType };
-  disabled?: ButtonKindType & { opacity?: OpacityType };
+  disabled?:
+    | (ButtonKindType & {
+        default?: ButtonKindType;
+        primary?: ButtonKindType;
+        secondary?: ButtonKindType;
+      })
+    | ({ [key: string]: ButtonKindType } & { opacity?: OpacityType });
```
Comment (jcfilben) [inline src/js/themes/base.d.ts:205]: Could we just have this?
```suggestion
  disabled?: ({ [key: string]: ButtonKindType } & { opacity?: OpacityType });
```

Comment (ShimiSun) [reply src/js/themes/base.d.ts:205]: This structure mimics the current structure for `active` and `hover`.

Comment (jcfilben) [reply src/js/themes/base.d.ts:205]: Ok I'm fine moving forward with this then

**Review by jcfilben — APPROVED** [review]
Looks good!

**Review by ericsoderberghp — APPROVED** [review]

**Pull request merged**